linux/setup_repo.sh

```diff
@@ -7,6 +7,26 @@ if [[ ${EUID} -ne 0 ]]; then
      exit 1
 fi
 
+# Check for apt version >= 2.4.0 (if found, assuming Debian-based compatible with repo)
+
+apt --version > /dev/null
+if [[ $? -eq 0 ]]; then
+    APT_VERSION=$(apt --version)
+    APT_MAJOR=$(echo "$APT_VERSION" | cut -d' ' -f 2 | cut -d'.' -f 1)
+    APT_MINOR=$(echo "$APT_VERSION" | cut -d' ' -f 2 | cut -d'.' -f 2)
+    echo "Apt version: ${APT_VERSION}"
+else
+    echo "This script is only compatible with Debian based distributions which have apt, but apt is not available. Please check that your OS is supported."
+    exit 1
+fi;
+
+if (( $(echo "${APT_MAJOR}.${APT_MINOR} < 2.4" | bc -l) )); then
```

Comment:
I'm not fully sure, but probably something like a "proceed anyway" would be good. Otherwise, we can add a comment here that this line can be removed to remove the check. That's useful for wrong detections.

Reply:
Added proceed anyway question